### ISSUE TITLE
support for arbitrary DB-API 2.0 (PEP 249) connections to `read_database`

### ISSUE THREAD

**Issue by antonysouthworth-halter** [pola-rs/polars#10416]
### Problem description

_Specifically_ I want to be able to use the [Trino Python client](https://github.com/trinodb/trino-python-client) but I think it would be useful more generally if arbitrary DB-API connections were accepted by `pl.read_database`. Since the DB-API is supposed to define the standard way to talk to databases it might also improve adoption of polars by being able to leverage existing DB-API implementations rather than waiting for `connectorx` or `adbc` to implement it.

It would look something like

```
from trino.dbapi import connect
import polars as pl

conn = connect(...)

df = pl.read_database("select * from sometable", conn=conn)
```

**Comment by alexander-beedie** [comment]
Yup, and we've already started on this; deprecated the previous `connection_uri` param name in favour of `connection` a few minor versions ago to start laying the groundwork for `pl.read_database` to accept user-supplied connection objects.

Will close this out as a dupe, in favour of the existing issue: https://github.com/pola-rs/polars/issues/10121.
(I'm thinking of tackling it all this weekend, so watch this space... ;)

**Comment by alexander-beedie** [comment]
@antonysouthworth-halter: Done; will be available in the upcoming `0.19.0` 👍
